# Global Energy & Climate Explorer

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import pandas as pd
pd.set_option('display.max_columns', None)
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go



# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/our-world-in-data-dataset-february-2026/owid-energy-data.csv
/kaggle/input/our-world-in-data-dataset-february-2026/owid-energy-codebook.csv


# Initial Data Inspection

In [2]:
df = pd.read_csv('/kaggle/input/our-world-in-data-dataset-february-2026/owid-energy-data.csv')
df.shape # IT HAS 23232 rows and 130 columns
info_df = pd.read_csv('/kaggle/input/our-world-in-data-dataset-february-2026/owid-energy-codebook.csv')

In [3]:
list(df.columns)

['country',
 'year',
 'iso_code',
 'population',
 'gdp',
 'biofuel_cons_change_pct',
 'biofuel_cons_change_twh',
 'biofuel_cons_per_capita',
 'biofuel_consumption',
 'biofuel_elec_per_capita',
 'biofuel_electricity',
 'biofuel_share_elec',
 'biofuel_share_energy',
 'carbon_intensity_elec',
 'coal_cons_change_pct',
 'coal_cons_change_twh',
 'coal_cons_per_capita',
 'coal_consumption',
 'coal_elec_per_capita',
 'coal_electricity',
 'coal_prod_change_pct',
 'coal_prod_change_twh',
 'coal_prod_per_capita',
 'coal_production',
 'coal_share_elec',
 'coal_share_energy',
 'electricity_demand',
 'electricity_demand_per_capita',
 'electricity_generation',
 'electricity_share_energy',
 'energy_cons_change_pct',
 'energy_cons_change_twh',
 'energy_per_capita',
 'energy_per_gdp',
 'fossil_cons_change_pct',
 'fossil_cons_change_twh',
 'fossil_elec_per_capita',
 'fossil_electricity',
 'fossil_energy_per_capita',
 'fossil_fuel_consumption',
 'fossil_share_elec',
 'fossil_share_energy',
 'gas_cons_chan

In [4]:
df.tail()

,country,year,iso_code,population,gdp,biofuel_cons_change_pct,biofuel_cons_change_twh,biofuel_cons_per_capita,biofuel_consumption,biofuel_elec_per_capita,biofuel_electricity,biofuel_share_elec,biofuel_share_energy,carbon_intensity_elec,coal_cons_change_pct,coal_cons_change_twh,coal_cons_per_capita,coal_consumption,coal_elec_per_capita,coal_electricity,coal_prod_change_pct,coal_prod_change_twh,coal_prod_per_capita,coal_production,coal_share_elec,coal_share_energy,electricity_demand,electricity_demand_per_capita,electricity_generation,electricity_share_energy,energy_cons_change_pct,energy_cons_change_twh,energy_per_capita,energy_per_gdp,fossil_cons_change_pct,fossil_cons_change_twh,fossil_elec_per_capita,fossil_electricity,fossil_energy_per_capita,fossil_fuel_consumption,fossil_share_elec,fossil_share_energy,gas_cons_change_pct,gas_cons_change_twh,gas_consumption,gas_elec_per_capita,gas_electricity,gas_energy_per_capita,gas_prod_change_pct,gas_prod_change_twh,gas_prod_per_capita,gas_production,gas_share_elec,gas_share_energy,greenhouse_gas_emissions,hydro_cons_change_pct,hydro_cons_change_twh,hydro_consumption,hydro_elec_per_capita,hydro_electricity,hydro_energy_per_capita,hydro_share_elec,hydro_share_energy,low_carbon_cons_change_pct,low_carbon_cons_change_twh,low_carbon_consumption,low_carbon_elec_per_capita,low_carbon_electricity,low_carbon_energy_per_capita,low_carbon_share_elec,low_carbon_share_energy,net_elec_imports,net_elec_imports_share_demand,nuclear_cons_change_pct,nuclear_cons_change_twh,nuclear_consumption,nuclear_elec_per_capita,nuclear_electricity,nuclear_energy_per_capita,nuclear_share_elec,nuclear_share_energy,oil_cons_change_pct,oil_cons_change_twh,oil_consumption,oil_elec_per_capita,oil_electricity,oil_energy_per_capita,oil_prod_change_pct,oil_prod_change_twh,oil_prod_per_capita,oil_production,oil_share_elec,oil_share_energy,other_renewable_consumption,other_renewable_electricity,other_renewable_exc_biofuel_electricity,other_renewables_cons_change_pct,other_renewables_cons_change_twh,other_renewables_elec_per_capita,other_renewables_elec_per_capita_exc_biofuel,other_renewables_energy_per_capita,other_renewables_share_elec,other_renewables_share_elec_exc_biofuel,other_renewables_share_energy,per_capita_electricity,primary_energy_consumption,renewables_cons_change_pct,renewables_cons_change_twh,renewables_consumption,renewables_elec_per_capita,renewables_electricity,renewables_energy_per_capita,renewables_share_elec,renewables_share_energy,solar_cons_change_pct,solar_cons_change_twh,solar_consumption,solar_elec_per_capita,solar_electricity,solar_energy_per_capita,solar_share_elec,solar_share_energy,wind_cons_change_pct,wind_cons_change_twh,wind_consumption,wind_elec_per_capita,wind_electricity,wind_energy_per_capita,wind_share_elec,wind_share_energy
23227,Zimbabwe,2020,ZWE,15526887.0,2.317871e+10,NaN,0.0,0.0,0.0,6.440,0.10,1.490,NaN,371.09,NaN,NaN,NaN,NaN,176.468,2.74,3.004,0.592,1306.886,20.292,40.835,NaN,8.69,559.674,6.71,NaN,-15.539,-6.179,2163.068,1.449,NaN,NaN,179.044,2.78,NaN,NaN,41.431,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,0.0,NaN,2.49,NaN,NaN,NaN,245.381,3.81,NaN,56.781,NaN,NaN,NaN,NaN,253.109,3.93,NaN,58.569,NaN,1.98,22.785,NaN,0.0,0.0,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,2.576,0.04,NaN,NaN,NaN,NaN,NaN,0.596,NaN,NaN,0.10,0.0,NaN,NaN,6.440,0.0,NaN,1.490,0.0,NaN,432.154,33.586,NaN,NaN,NaN,253.109,3.93,NaN,58.569,NaN,NaN,NaN,NaN,1.288,0.02,NaN,0.298,NaN,NaN,NaN,NaN,0.0,0.0,NaN,0.0,NaN
23228,Zimbabwe,2021,ZWE,15797220.0,2.514009e+10,NaN,0.0,0.0,0.0,6.963,0.11,1.284,NaN,269.55,NaN,NaN,NaN,NaN,158.889,2.51,19.708,3.999,1537.673,24.291,29.288,NaN,10.31,652.646,8.57,NaN,12.059,4.050,2382.434,1.497,NaN,NaN,158.889,2.51,NaN,NaN,29.288,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,0.0,NaN,2.31,NaN,NaN,NaN,375.383,5.93,NaN,69.195,NaN,NaN,NaN,NaN,383.612,6.06,NaN,70.712,NaN,1.74,16.877,NaN,0.0,0.0,0.0,0.0,0.0,0.0,NaN,NaN,NaN,NaN,0.000,0.00,NaN,NaN,NaN,NaN,NaN,0.000,NaN,NaN,0.11,0.0,NaN,NaN,6.963,0.0,NaN,1.284,0.0,NaN

In [5]:
dict(df[df['country'] == 'Philippines'].isnull().sum())

{'country': np.int64(0),
 'year': np.int64(0),
 'iso_code': np.int64(0),
 'population': np.int64(0),
 'gdp': np.int64(9),
 'biofuel_cons_change_pct': np.int64(105),
 'biofuel_cons_change_twh': np.int64(66),
 'biofuel_cons_per_capita': np.int64(65),
 'biofuel_consumption': np.int64(65),
 'biofuel_elec_per_capita': np.int64(100),
 'biofuel_electricity': np.int64(100),
 'biofuel_share_elec': np.int64(100),
 'biofuel_share_energy': np.int64(65),
 'carbon_intensity_elec': np.int64(100),
 'coal_cons_change_pct': np.int64(66),
 'coal_cons_change_twh': np.int64(66),
 'coal_cons_per_capita': np.int64(65),
 'coal_consumption': np.int64(65),
 'coal_elec_per_capita': np.int64(100),
 'coal_electricity': np.int64(100),
 'coal_prod_change_pct': np.int64(21),
 'coal_prod_change_twh': np.int64(9),
 'coal_prod_per_capita': np.int64(8),
 'coal_production': np.int64(8),
 'coal_share_elec': np.int64(100),
 'coal_share_energy': np.int64(65),
 'electricity_demand': np.int64(100),
 'electricity_demand_per_cap

In [6]:
philippines = df[df['country']=='Philippines']
philippines

,country,year,iso_code,population,gdp,biofuel_cons_change_pct,biofuel_cons_change_twh,biofuel_cons_per_capita,biofuel_consumption,biofuel_elec_per_capita,biofuel_electricity,biofuel_share_elec,biofuel_share_energy,carbon_intensity_elec,coal_cons_change_pct,coal_cons_change_twh,coal_cons_per_capita,coal_consumption,coal_elec_per_capita,coal_electricity,coal_prod_change_pct,coal_prod_change_twh,coal_prod_per_capita,coal_production,coal_share_elec,coal_share_energy,electricity_demand,electricity_demand_per_capita,electricity_generation,electricity_share_energy,energy_cons_change_pct,energy_cons_change_twh,energy_per_capita,energy_per_gdp,fossil_cons_change_pct,fossil_cons_change_twh,fossil_elec_per_capita,fossil_electricity,fossil_energy_per_capita,fossil_fuel_consumption,fossil_share_elec,fossil_share_energy,gas_cons_change_pct,gas_cons_change_twh,gas_consumption,gas_elec_per_capita,gas_electricity,gas_energy_per_capita,gas_prod_change_pct,gas_prod_change_twh,gas_prod_per_capita,gas_production,gas_share_elec,gas_share_energy,greenhouse_gas_emissions,hydro_cons_change_pct,hydro_cons_change_twh,hydro_consumption,hydro_elec_per_capita,hydro_electricity,hydro_energy_per_capita,hydro_share_elec,hydro_share_energy,low_carbon_cons_change_pct,low_carbon_cons_change_twh,low_carbon_consumption,low_carbon_elec_per_capita,low_carbon_electricity,low_carbon_energy_per_capita,low_carbon_share_elec,low_carbon_share_energy,net_elec_imports,net_elec_imports_share_demand,nuclear_cons_change_pct,nuclear_cons_change_twh,nuclear_consumption,nuclear_elec_per_capita,nuclear_electricity,nuclear_energy_per_capita,nuclear_share_elec,nuclear_share_energy,oil_cons_change_pct,oil_cons_change_twh,oil_consumption,oil_elec_per_capita,oil_electricity,oil_energy_per_capita,oil_prod_change_pct,oil_prod_change_twh,oil_prod_per_capita,oil_production,oil_share_elec,oil_share_energy,other_renewable_consumption,other_renewable_electricity,other_renewable_exc_biofuel_electricity,other_renewables_cons_change_pct,other_renewables_cons_change_twh,other_renewables_elec_per_capita,other_renewables_elec_per_capita_exc_biofuel,other_renewables_energy_per_capita,other_renewables_share_elec,other_renewables_share_elec_exc_biofuel,other_renewables_share_energy,per_capita_electricity,primary_energy_consumption,renewables_cons_change_pct,renewables_cons_change_twh,renewables_consumption,renewables_elec_per_capita,renewables_electricity,renewables_energy_per_capita,renewables_share_elec,renewables_share_energy,solar_cons_change_pct,solar_cons_change_twh,solar_consumption,solar_elec_per_capita,solar_electricity,solar_energy_per_capita,solar_share_elec,solar_share_energy,wind_cons_change_pct,wind_cons_change_twh,wind_consumption,wind_elec_per_capita,wind_electricity,wind_energy_per_capita,wind_share_elec,wind_share_energy
17257,Philippines,1900,PHL,6644021.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
17258,Philippines,1901,PHL,6739540.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
17259,Philippines,1902,PHL,6845501.0,8.164457e+09,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,N

In [7]:
year = df.groupby('year').size()
dict(year)

{1900: np.int64(118),
 1901: np.int64(118),
 1902: np.int64(118),
 1903: np.int64(118),
 1904: np.int64(118),
 1905: np.int64(118),
 1906: np.int64(118),
 1907: np.int64(118),
 1908: np.int64(118),
 1909: np.int64(118),
 1910: np.int64(118),
 1911: np.int64(118),
 1912: np.int64(118),
 1913: np.int64(118),
 1914: np.int64(118),
 1915: np.int64(118),
 1916: np.int64(118),
 1917: np.int64(118),
 1918: np.int64(118),
 1919: np.int64(118),
 1920: np.int64(118),
 1921: np.int64(118),
 1922: np.int64(118),
 1923: np.int64(118),
 1924: np.int64(118),
 1925: np.int64(118),
 1926: np.int64(118),
 1927: np.int64(118),
 1928: np.int64(118),
 1929: np.int64(118),
 1930: np.int64(118),
 1931: np.int64(118),
 1932: np.int64(118),
 1933: np.int64(118),
 1934: np.int64(118),
 1935: np.int64(118),
 1936: np.int64(118),
 1937: np.int64(118),
 1938: np.int64(118),
 1939: np.int64(118),
 1940: np.int64(118),
 1941: np.int64(118),
 1942: np.int64(118),
 1943: np.int64(118),
 1944: np.int64(118),
 1945: np.

In [8]:
unique_countries = df['country'].nunique()
unique_countries

314

In [9]:
# FINDING
sea_countries = ['Philippines', 'Indonesia', 'Vietnam', 
                 'Thailand', 'Myanmar', 'East Timor', 
                 'Malaysia', 'Singapore', 'Cambodia',
                'Laos', 'Brunei']
sea_df = df[df['country'].isin(sea_countries)]
sea_df


,country,year,iso_code,population,gdp,biofuel_cons_change_pct,biofuel_cons_change_twh,biofuel_cons_per_capita,biofuel_consumption,biofuel_elec_per_capita,biofuel_electricity,biofuel_share_elec,biofuel_share_energy,carbon_intensity_elec,coal_cons_change_pct,coal_cons_change_twh,coal_cons_per_capita,coal_consumption,coal_elec_per_capita,coal_electricity,coal_prod_change_pct,coal_prod_change_twh,coal_prod_per_capita,coal_production,coal_share_elec,coal_share_energy,electricity_demand,electricity_demand_per_capita,electricity_generation,electricity_share_energy,energy_cons_change_pct,energy_cons_change_twh,energy_per_capita,energy_per_gdp,fossil_cons_change_pct,fossil_cons_change_twh,fossil_elec_per_capita,fossil_electricity,fossil_energy_per_capita,fossil_fuel_consumption,fossil_share_elec,fossil_share_energy,gas_cons_change_pct,gas_cons_change_twh,gas_consumption,gas_elec_per_capita,gas_electricity,gas_energy_per_capita,gas_prod_change_pct,gas_prod_change_twh,gas_prod_per_capita,gas_production,gas_share_elec,gas_share_energy,greenhouse_gas_emissions,hydro_cons_change_pct,hydro_cons_change_twh,hydro_consumption,hydro_elec_per_capita,hydro_electricity,hydro_energy_per_capita,hydro_share_elec,hydro_share_energy,low_carbon_cons_change_pct,low_carbon_cons_change_twh,low_carbon_consumption,low_carbon_elec_per_capita,low_carbon_electricity,low_carbon_energy_per_capita,low_carbon_share_elec,low_carbon_share_energy,net_elec_imports,net_elec_imports_share_demand,nuclear_cons_change_pct,nuclear_cons_change_twh,nuclear_consumption,nuclear_elec_per_capita,nuclear_electricity,nuclear_energy_per_capita,nuclear_share_elec,nuclear_share_energy,oil_cons_change_pct,oil_cons_change_twh,oil_consumption,oil_elec_per_capita,oil_electricity,oil_energy_per_capita,oil_prod_change_pct,oil_prod_change_twh,oil_prod_per_capita,oil_production,oil_share_elec,oil_share_energy,other_renewable_consumption,other_renewable_electricity,other_renewable_exc_biofuel_electricity,other_renewables_cons_change_pct,other_renewables_cons_change_twh,other_renewables_elec_per_capita,other_renewables_elec_per_capita_exc_biofuel,other_renewables_energy_per_capita,other_renewables_share_elec,other_renewables_share_elec_exc_biofuel,other_renewables_share_energy,per_capita_electricity,primary_energy_consumption,renewables_cons_change_pct,renewables_cons_change_twh,renewables_consumption,renewables_elec_per_capita,renewables_electricity,renewables_energy_per_capita,renewables_share_elec,renewables_share_energy,solar_cons_change_pct,solar_cons_change_twh,solar_consumption,solar_elec_per_capita,solar_electricity,solar_energy_per_capita,solar_share_elec,solar_share_energy,wind_cons_change_pct,wind_cons_change_twh,wind_consumption,wind_elec_per_capita,wind_electricity,wind_energy_per_capita,wind_share_elec,wind_share_energy
3140,Brunei,1900,BRN,16978.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3141,Brunei,1901,BRN,17540.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000,0.000,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.000,0.000,0.000,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3142,Brunei,1902,BRN,18120.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Na

In [10]:
null_sea = sea_df.isnull().groupby(sea_df['country']).sum()
null_sea

,country,year,iso_code,population,gdp,biofuel_cons_change_pct,biofuel_cons_change_twh,biofuel_cons_per_capita,biofuel_consumption,biofuel_elec_per_capita,biofuel_electricity,biofuel_share_elec,biofuel_share_energy,carbon_intensity_elec,coal_cons_change_pct,coal_cons_change_twh,coal_cons_per_capita,coal_consumption,coal_elec_per_capita,coal_electricity,coal_prod_change_pct,coal_prod_change_twh,coal_prod_per_capita,coal_production,coal_share_elec,coal_share_energy,electricity_demand,electricity_demand_per_capita,electricity_generation,electricity_share_energy,energy_cons_change_pct,energy_cons_change_twh,energy_per_capita,energy_per_gdp,fossil_cons_change_pct,fossil_cons_change_twh,fossil_elec_per_capita,fossil_electricity,fossil_energy_per_capita,fossil_fuel_consumption,fossil_share_elec,fossil_share_energy,gas_cons_change_pct,gas_cons_change_twh,gas_consumption,gas_elec_per_capita,gas_electricity,gas_energy_per_capita,gas_prod_change_pct,gas_prod_change_twh,gas_prod_per_capita,gas_production,gas_share_elec,gas_share_energy,greenhouse_gas_emissions,hydro_cons_change_pct,hydro_cons_change_twh,hydro_consumption,hydro_elec_per_capita,hydro_electricity,hydro_energy_per_capita,hydro_share_elec,hydro_share_energy,low_carbon_cons_change_pct,low_carbon_cons_change_twh,low_carbon_consumption,low_carbon_elec_per_capita,low_carbon_electricity,low_carbon_energy_per_capita,low_carbon_share_elec,low_carbon_share_energy,net_elec_imports,net_elec_imports_share_demand,nuclear_cons_change_pct,nuclear_cons_change_twh,nuclear_consumption,nuclear_elec_per_capita,nuclear_electricity,nuclear_energy_per_capita,nuclear_share_elec,nuclear_share_energy,oil_cons_change_pct,oil_cons_change_twh,oil_consumption,oil_elec_per_capita,oil_electricity,oil_energy_per_capita,oil_prod_change_pct,oil_prod_change_twh,oil_prod_per_capita,oil_production,oil_share_elec,oil_share_energy,other_renewable_consumption,other_renewable_electricity,other_renewable_exc_biofuel_electricity,other_renewables_cons_change_pct,other_renewables_cons_change_twh,other_renewables_elec_per_capita,other_renewables_elec_per_capita_exc_biofuel,other_renewables_energy_per_capita,other_renewables_share_elec,other_renewables_share_elec_exc_biofuel,other_renewables_share_energy,per_capita_electricity,primary_energy_consumption,renewables_cons_change_pct,renewables_cons_change_twh,renewables_consumption,renewables_elec_per_capita,renewables_electricity,renewables_energy_per_capita,renewables_share_elec,renewables_share_energy,solar_cons_change_pct,solar_cons_change_twh,solar_consumption,solar_elec_per_capita,solar_electricity,solar_energy_per_capita,solar_share_elec,solar_share_energy,wind_cons_change_pct,wind_cons_change_twh,wind_consumption,wind_elec_per_capita,wind_electricity,wind_energy_per_capita,wind_share_elec,wind_share_energy
country,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
Brunei,0,0,0,0,125,125,66,65,65,101,101,101,125,101,125,125,125,125,101,101,125,89,88,88,101,125,101,101,101,125,82,82,81,125,125,125,101,101,125,125,101,125,125,125,125,101,101,125,48,1,0,0,101,125,101,125,125,125,101,101,125,101,125,125,125,125,101,101,125,101,125,101,101,125,66,65,65,65,65,101,125,125,125,125,101,101,125,32,1,0,0,101,125,125,101,101,125,125,101,101,125,101,101,125,101,81,125,125,125,101,101,125,101,125,125,125,125,101,101,125,101,125,125,125,125,101,101,125,101,125
Cambodia,0,0,0,0,2,45,45,45,45,20,20,20,45,20,45,45,45,45,20,20,45,9,8,8,20,45,20,20,20,45,2,2,1,2,45,45,20,20,45,45,20,45,45,45,45,20,20,45,45,9,8,8,20,45,20,45,45,45,20,20,45,20,45,45,45,45,20,20,45,20,45,20,20,45,45,45,20,20,45,20,45,45,45,45,20,20,45,45,9,8,8,20,45,45,20,20,45,45,20,20,45,20,20,45,20,1,45,45,45,20,20,45,20,45,45,45,45,20,20,45,20,45,45,45,45,20,20,45,20,45
East Timor,0,0,0,0,21,21,21,21,21,0,0,0,21,0,21,21,21,21,0,0,21,8,7,7,0,21,0,0,0,21,1,1,0,21,21,21,0,0,21,21,0,21,21,21,21,0,0,21,11,8,7,7,0,21,0,21,21,21,0,0,21,0,21,21,21,21,0

In [11]:
rows_sea = sea_df[sea_df['year'] >= 2004].groupby('country')['year'].size()
rows_sea



country
Brunei         21
Cambodia       21
East Timor     20
Indonesia      21
Laos           20
Malaysia       21
Myanmar        21
Philippines    21
Singapore      21
Thailand       21
Vietnam        21
Name: year, dtype: int64

In [12]:
sea_df = sea_df[sea_df['year'] >= 2005]
sea_df.shape

(218, 130)

In [13]:
dict(sea_df.isnull().sum())

{'country': np.int64(0),
 'year': np.int64(0),
 'iso_code': np.int64(0),
 'population': np.int64(0),
 'gdp': np.int64(56),
 'biofuel_cons_change_pct': np.int64(142),
 'biofuel_cons_change_twh': np.int64(58),
 'biofuel_cons_per_capita': np.int64(58),
 'biofuel_consumption': np.int64(58),
 'biofuel_elec_per_capita': np.int64(1),
 'biofuel_electricity': np.int64(1),
 'biofuel_share_elec': np.int64(1),
 'biofuel_share_energy': np.int64(98),
 'carbon_intensity_elec': np.int64(1),
 'coal_cons_change_pct': np.int64(98),
 'coal_cons_change_twh': np.int64(98),
 'coal_cons_per_capita': np.int64(98),
 'coal_consumption': np.int64(98),
 'coal_elec_per_capita': np.int64(1),
 'coal_electricity': np.int64(1),
 'coal_prod_change_pct': np.int64(110),
 'coal_prod_change_twh': np.int64(62),
 'coal_prod_per_capita': np.int64(62),
 'coal_production': np.int64(62),
 'coal_share_elec': np.int64(1),
 'coal_share_energy': np.int64(98),
 'electricity_demand': np.int64(1),
 'electricity_demand_per_capita': np.in

In [14]:
null_sea = sea_df.isnull().groupby(sea_df['country']).sum()
null_sea

,country,year,iso_code,population,gdp,biofuel_cons_change_pct,biofuel_cons_change_twh,biofuel_cons_per_capita,biofuel_consumption,biofuel_elec_per_capita,biofuel_electricity,biofuel_share_elec,biofuel_share_energy,carbon_intensity_elec,coal_cons_change_pct,coal_cons_change_twh,coal_cons_per_capita,coal_consumption,coal_elec_per_capita,coal_electricity,coal_prod_change_pct,coal_prod_change_twh,coal_prod_per_capita,coal_production,coal_share_elec,coal_share_energy,electricity_demand,electricity_demand_per_capita,electricity_generation,electricity_share_energy,energy_cons_change_pct,energy_cons_change_twh,energy_per_capita,energy_per_gdp,fossil_cons_change_pct,fossil_cons_change_twh,fossil_elec_per_capita,fossil_electricity,fossil_energy_per_capita,fossil_fuel_consumption,fossil_share_elec,fossil_share_energy,gas_cons_change_pct,gas_cons_change_twh,gas_consumption,gas_elec_per_capita,gas_electricity,gas_energy_per_capita,gas_prod_change_pct,gas_prod_change_twh,gas_prod_per_capita,gas_production,gas_share_elec,gas_share_energy,greenhouse_gas_emissions,hydro_cons_change_pct,hydro_cons_change_twh,hydro_consumption,hydro_elec_per_capita,hydro_electricity,hydro_energy_per_capita,hydro_share_elec,hydro_share_energy,low_carbon_cons_change_pct,low_carbon_cons_change_twh,low_carbon_consumption,low_carbon_elec_per_capita,low_carbon_electricity,low_carbon_energy_per_capita,low_carbon_share_elec,low_carbon_share_energy,net_elec_imports,net_elec_imports_share_demand,nuclear_cons_change_pct,nuclear_cons_change_twh,nuclear_consumption,nuclear_elec_per_capita,nuclear_electricity,nuclear_energy_per_capita,nuclear_share_elec,nuclear_share_energy,oil_cons_change_pct,oil_cons_change_twh,oil_consumption,oil_elec_per_capita,oil_electricity,oil_energy_per_capita,oil_prod_change_pct,oil_prod_change_twh,oil_prod_per_capita,oil_production,oil_share_elec,oil_share_energy,other_renewable_consumption,other_renewable_electricity,other_renewable_exc_biofuel_electricity,other_renewables_cons_change_pct,other_renewables_cons_change_twh,other_renewables_elec_per_capita,other_renewables_elec_per_capita_exc_biofuel,other_renewables_energy_per_capita,other_renewables_share_elec,other_renewables_share_elec_exc_biofuel,other_renewables_share_energy,per_capita_electricity,primary_energy_consumption,renewables_cons_change_pct,renewables_cons_change_twh,renewables_consumption,renewables_elec_per_capita,renewables_electricity,renewables_energy_per_capita,renewables_share_elec,renewables_share_energy,solar_cons_change_pct,solar_cons_change_twh,solar_consumption,solar_elec_per_capita,solar_electricity,solar_energy_per_capita,solar_share_elec,solar_share_energy,wind_cons_change_pct,wind_cons_change_twh,wind_consumption,wind_elec_per_capita,wind_electricity,wind_energy_per_capita,wind_share_elec,wind_share_energy
country,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
Brunei,0,0,0,0,20,20,0,0,0,1,1,1,20,1,20,20,20,20,1,1,20,8,8,8,1,20,1,1,1,20,1,1,1,20,20,20,1,1,20,20,1,20,20,20,20,1,1,20,0,0,0,0,1,20,1,20,20,20,1,1,20,1,20,20,20,20,1,1,20,1,20,1,1,20,0,0,0,0,0,1,20,20,20,20,1,1,20,0,0,0,0,1,20,20,1,1,20,20,1,1,20,1,1,20,1,1,20,20,20,1,1,20,1,20,20,20,20,1,1,20,1,20,20,20,20,1,1,20,1,20
Cambodia,0,0,0,0,2,20,20,20,20,0,0,0,20,0,20,20,20,20,0,0,20,8,8,8,0,20,0,0,0,20,1,1,1,2,20,20,0,0,20,20,0,20,20,20,20,0,0,20,20,8,8,8,0,20,0,20,20,20,0,0,20,0,20,20,20,20,0,0,20,0,20,0,0,20,20,20,0,0,20,0,20,20,20,20,0,0,20,20,8,8,8,0,20,20,0,0,20,20,0,0,20,0,0,20,0,1,20,20,20,0,0,20,0,20,20,20,20,0,0,20,0,20,20,20,20,0,0,20,0,20
East Timor,0,0,0,0,19,19,19,19,19,0,0,0,19,0,19,19,19,19,0,0,19,7,7,7,0,19,0,0,0,19,0,0,0,19,19,19,0,0,19,19,0,19,19,19,19,0,0,19,9,7,7,7,0,19,0,19,19,19,0,0,19,0,19,19,19,19,0,0,19,0,19,0,0,19,19,19,0,0,19,0,19,19,19,19,0,0,19,7,7,7,7,0,19,19,0,0,19,19,0,0,19,0,0,19,0,0,19,19,19,0,0,19,0,19,19,19,19,0,0,19,0,19,19,19,19,0,0,19,0,19
Indonesia,0,0,0,0,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,

In [15]:
selected_columns = [
    'country', 'year', 'iso_code',
    'population', 'gdp',
    'fossil_share_energy',
    'renewables_share_energy',
    'solar_share_energy',
    'wind_share_energy',
    'hydro_share_energy',
    'low_carbon_share_energy',
    'primary_energy_consumption',
    'fossil_fuel_consumption',
    'renewables_consumption',
    'energy_per_capita',
    'renewables_cons_change_pct',
    'fossil_cons_change_pct',
    'greenhouse_gas_emissions',
    'carbon_intensity_elec',
    'energy_per_gdp'
]

f_sea_df = sea_df[selected_columns].copy()
f_sea_df.shape


(218, 20)

In [16]:
nulls = f_sea_df.isnull().groupby(sea_df['country']).sum()
nulls

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp
country,,,,,,,,,,,,,,,,,,,,
Brunei,0,0,0,0,20,20,20,20,20,20,20,1,20,20,1,20,20,1,1,20
Cambodia,0,0,0,0,2,20,20,20,20,20,20,1,20,20,1,20,20,0,0,2
East Timor,0,0,0,0,19,19,19,19,19,19,19,0,19,19,0,19,19,0,0,19
Indonesia,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Laos,0,0,0,0,1,19,19,19,19,19,19,0,19,19,0,19,19,0,0,1
Malaysia,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Myanmar,0,0,0,0,2,20,20,20,20,20,20,1,20,20,1,20,20,0,0,2
Philippines,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Singapore,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2


In [17]:
# FINDING
top_sea = ['Philippines', 'Indonesia', 'Vietnam', 
                 'Thailand', 
                 'Malaysia', 'Singapore']
sea_six_df = f_sea_df[f_sea_df['country'].isin(top_sea)]
sea_six_df

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp
9739,Indonesia,2005,IDN,230871648.0,1.474875e+12,96.506,3.494,0.000,0.000,2.034,3.494,1416.892,1367.980,49.523,6137.142,6.250,2.824,86.45,650.54,0.961
9740,Indonesia,2006,IDN,233951650.0,1.565013e+12,96.766,3.234,0.000,0.000,1.784,3.234,1441.128,1395.470,46.632,6159.938,-5.862,2.010,92.20,668.12,0.921
9741,Indonesia,2007,IDN,237062338.0,1.674227e+12,96.626,3.374,0.000,0.000,1.939,3.374,1546.681,1495.455,52.218,6524.362,13.382,7.165,98.41,663.14,0.924
9742,Indonesia,2008,IDN,240157902.0,1.809491e+12,96.380,3.620,0.000,0.000,1.942,3.620,1564.584,1509.474,56.691,6514.816,8.398,0.937,101.80,652.98,0.865
9743,Indonesia,2009,IDN,243220024.0,1.905717e+12,96.264,3.736,0.000,0.001,1.856,3.736,1607.271,1549.363,60.130,6608.300,8.580,2.643,107.60,660.65,0.843
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
22475,Vietnam,2020,VNM,98079196.0,7.526296e+11,82.724,17.276,2.186,0.199,14.803,17.276,1228.728,1005.433,209.976,12527.920,17.971,-1.251,128.94,528.44,1.633
22476,Vietnam,2021,VNM,98935101.0,7.719120e+11,77.840,22.160,5.338,0.682,16.058,22.160,1223.385,939.331,267.416,12365.529,27.705,-6.574,120.48,475.06,1.585
22477,Vietnam,2022,VNM,99680656.0,8.338039e+11,74.688,25.312,4.967,1.753,18.499,25.312,1293.495,951.403,322.437,12976.387,20.873,1.285,110.13,415.41,1.551
22478,Vietnam,2023,VNM,100352189.0,NaN,78.612,21.388,4.617,2.082,14.489,21.388,1375.936,1066.903,290.271,13711.071,-9.427,12.140,130.66,472.40,NaN


In [18]:
sea_six_df.isnull().groupby(sea_df['country']).sum()

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp
country,,,,,,,,,,,,,,,,,,,,
Indonesia,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Malaysia,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Philippines,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Singapore,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Thailand,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2
Vietnam,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2


In [19]:
# Year-over-Year (YoY) change

sea_six_df['yoy_rse'] = sea_six_df.sort_values('year').groupby('country')['renewables_share_energy'].pct_change() * 100
sea_six_df['yoy_fse'] = sea_six_df.sort_values('year').groupby('country')['fossil_share_energy'].pct_change() * 100


/tmp/ipykernel_17/4082330482.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  sea_six_df['yoy_rse'] = sea_six_df.sort_values('year').groupby('country')['renewables_share_energy'].pct_change() * 100
/tmp/ipykernel_17/4082330482.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  sea_six_df['yoy_fse'] = sea_six_df.sort_values('year').groupby('country')['fossil_share_energy'].pct_change() * 100


In [20]:
# Cumulative Change Since Baseline
sea_six_df = sea_six_df.sort_values('year')

sea_six_df['cum_rse'] = sea_six_df.groupby('country')['renewables_share_energy'].transform(lambda x: (x - x.iloc[0]) / x.iloc[0] * 100)
sea_six_df['cum_fse'] = sea_six_df.groupby('country')['fossil_share_energy'].transform(lambda x: (x - x.iloc[0]) / x.iloc[0] * 100)


In [21]:
sea_six_df.tail(20)

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp,yoy_rse,yoy_fse,cum_rse,cum_fse
18759,Singapore,2021,SGP,5546291.0,4.455520e+11,99.662,0.338,0.107,0.000,0.000,0.338,912.046,908.861,3.080,164442.578,9.140,-0.170,28.24,506.18,2.047,7.643312,-0.024076,22.909091,-0.063174
20546,Thailand,2021,THA,71727338.0,1.095187e+12,92.966,7.034,0.888,0.629,0.808,7.034,1386.818,1286.562,97.350,19334.578,-2.550,0.447,99.83,566.06,1.266,0.918221,-0.068795,233.681214,-5.032076
20547,Thailand,2022,THA,71735321.0,1.124144e+12,92.752,7.248,0.892,0.530,1.171,7.248,1387.880,1284.390,100.366,19347.238,-0.272,-0.169,101.06,560.29,1.235,3.042366,-0.230192,243.833017,-5.250684
11966,Malaysia,2022,MYS,34695494.0,8.775842e+11,92.167,7.833,0.508,0.000,5.957,7.833,1355.509,1244.331,105.746,39068.727,15.421,8.121,110.92,604.04,1.545,2.445723,-0.202482,263.480278,-5.803056
9756,Indonesia,2022,IDN,278830527.0,3.500094e+12,89.792,10.208,0.040,0.032,2.435,10.208,2749.242,2472.905,281.127,9859.904,13.991,26.317,225.41,675.81,0.785,-7.836764,0.976114,192.157985,-6.957080
22477,Vietnam,2022,VNM,99680656.0,8.338039e+11,74.688,25.312,4.967,1.753,18.499,25.312,1293.495,951.403,322.437,12976.387,20.873,1.285,110.13,415.41,1.551,14.223827,-4.049332,108.192137,-14.974614
17379,Philippines,2022,PHL,113964343.0,1.014811e+12,86.968,13.032,0.781,0.450,4.324,13.032,570.979,500.444,74.990,5010.157,6.177,4.568,67.21,602.67,0.563,0.648749,-0.096494,-21.743830,4.344488
18760,Singapore,2022,SGP,5649888.0,4.618011e+11,99.479,0.521,0.183,0.000,0.000,0.521,895.630,890.796,4.668,158521.719,53.019,-1.988,28.70,504.22,1.939,54.142012,-0.183621,89.454545,-0.246678
9757,Indonesia,2023,IDN,281190068.0,NaN,89.305,10.695,0.062,0.042,2.149,10.695,2785.135,2491.822,298.421,9904.812,10.337,0.765,238.69,680.80,NaN,4.770768,-0.542365,206.096165,-7.461712
17380,Philippines,2023,PHL,114891199.0,NaN,87.078,12.922,1.014,0.529,4.109,12.922,608.771,533.969,79.238,5298.678,6.168,6.699,71.52,612.54,NaN,-0.844076,0.126483,-22.404372,4.476466


In [22]:
info_df[info_df['column']=='energy_per_gdp']

,column,title,description,unit,source
33,energy_per_gdp,Primary energy consumption per GDP,Measured in kilowatt-hours per international-$.,kilowatt-hours per $ (kWh),U.S. Energy Information Administration - Inter...


In [23]:
sea_six_df['energy_gap'] = sea_six_df['fossil_share_energy'] - sea_six_df['renewables_share_energy']
sea_six_df

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp,yoy_rse,yoy_fse,cum_rse,cum_fse,energy_gap
9739,Indonesia,2005,IDN,230871648.0,1.474875e+12,96.506,3.494,0.000,0.000,2.034,3.494,1416.892,1367.980,49.523,6137.142,6.250,2.824,86.45,650.54,0.961,NaN,NaN,0.000000,0.000000,93.012
11949,Malaysia,2005,MYS,25836075.0,3.997810e+11,97.845,2.155,0.000,0.000,1.933,2.155,836.425,817.333,17.999,32374.299,7.320,8.252,64.43,652.46,2.092,NaN,NaN,0.000000,0.000000,95.690
18743,Singapore,2005,SGP,4268363.0,1.906915e+11,99.725,0.275,0.000,0.000,0.000,0.275,542.302,540.778,1.493,127051.617,-0.280,5.622,20.85,545.53,2.844,NaN,NaN,0.000000,0.000000,99.450
17362,Philippines,2005,PHL,88015959.0,4.268641e+11,83.347,16.653,0.001,0.015,7.010,16.653,319.341,268.047,53.558,3628.220,-2.919,0.744,25.38,448.65,0.748,NaN,NaN,0.000000,0.000000,66.694
22460,Vietnam,2005,VNM,81088309.0,3.089887e+11,87.842,12.158,0.000,0.000,12.115,12.158,369.791,322.285,44.605,4560.353,-6.920,8.722,24.09,462.56,1.197,NaN,NaN,0.000000,0.000000,75.684
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11968,Malaysia,2024,MYS,35557670.0,NaN,91.718,8.282,0.624,0.000,6.094,8.282,1381.980,1262.247,113.976,38865.867,7.110,4.122,119.81,604.43,NaN,2.032771,-0.179576,284.315545,-6.261945,83.436
17381,Philippines,2024,PHL,115843663.0,NaN,87.182,12.818,1.408,0.482,4.130,12.818,652.563,572.515,84.178,5633.132,6.765,7.219,77.07,612.35,NaN,-0.804829,0.119433,-23.028884,4.601245,74.364
18762,Singapore,2024,SGP,5832387.0,NaN,99.455,0.545,0.295,0.000,0.000,0.545,1053.921,1047.929,5.745,180701.453,19.774,6.255,29.73,498.74,NaN,10.547667,-0.052258,98.181818,-0.270745,98.910
20549,Thailand,2024,THA,71668010.0,NaN,92.597,7.403,0.935,0.582,1.104,7.403,1420.819,1312.641,104.940,19825.014,-0.549,2.597,110.68,554.73,NaN,-2.694532,0.221881,251.185958,-5.409022,85.194


In [24]:
energy_gap_pivot = sea_six_df.pivot_table(index='year', columns='country', values='energy_gap')
energy_gap_pivot

country,Indonesia,Malaysia,Philippines,Singapore,Thailand,Vietnam
year,,,,,,
2005,93.012,95.690,66.694,99.450,95.784,75.684
2006,93.532,95.516,61.364,99.506,94.552,68.468
2007,93.252,95.874,65.318,99.524,94.336,67.302
2008,92.760,94.946,63.296,99.528,94.276,69.636
2009,92.528,95.024,64.166,99.554,94.040,65.892
2010,91.166,95.596,68.666,99.534,94.500,72.138
2011,92.892,94.594,66.252,99.538,93.116,63.842
2012,92.846,94.246,65.646,99.516,92.438,55.024
2013,90.932,93.344,69.014,99.470,92.400,54.130


In [25]:
66.694 - 74.364

-7.670000000000002

In [26]:
# yoy_rse	yoy_fse	cum_rse	cum_fse
random = sea_six_df.pivot_table(index='year', columns='country', values='cum_rse')
random

country,Indonesia,Malaysia,Philippines,Singapore,Thailand,Vietnam
year,,,,,,
2005,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
2006,-7.441328,4.037123,16.003123,-10.181818,29.222011,29.675934
2007,-3.434459,-4.269142,4.131388,-13.454545,34.345351,34.471130
2008,3.606182,17.262181,10.202366,-14.181818,35.768501,24.872512
2009,6.926159,15.452436,7.590224,-18.909091,41.366224,40.269781
2010,26.416714,2.180974,-5.920855,-15.272727,30.455408,14.582991
2011,1.717230,25.429234,1.327088,-16.000000,63.282732,48.700444
2012,2.375501,33.503480,3.146580,-12.000000,79.364326,84.964632
2013,29.765312,54.431555,-6.965712,-3.636364,80.265655,88.641224


In [27]:
bins = [0, 13889, 41667, float('inf')]

# INFORMATION ABOUT THE BINS
# Since Energy Per Capita is Measured in kilowatt-hours per person. 
# The following value of bins is defined in kilowatt-hours that 
# can be converted into Gigajoules

labels = ['Low', 'Medium', 'High']
sea_six_df['category'] = pd.cut(sea_six_df['energy_per_capita'], bins=bins, labels=labels)
sea_six_df


,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp,yoy_rse,yoy_fse,cum_rse,cum_fse,energy_gap,category
9739,Indonesia,2005,IDN,230871648.0,1.474875e+12,96.506,3.494,0.000,0.000,2.034,3.494,1416.892,1367.980,49.523,6137.142,6.250,2.824,86.45,650.54,0.961,NaN,NaN,0.000000,0.000000,93.012,Low
11949,Malaysia,2005,MYS,25836075.0,3.997810e+11,97.845,2.155,0.000,0.000,1.933,2.155,836.425,817.333,17.999,32374.299,7.320,8.252,64.43,652.46,2.092,NaN,NaN,0.000000,0.000000,95.690,Medium
18743,Singapore,2005,SGP,4268363.0,1.906915e+11,99.725,0.275,0.000,0.000,0.000,0.275,542.302,540.778,1.493,127051.617,-0.280,5.622,20.85,545.53,2.844,NaN,NaN,0.000000,0.000000,99.450,High
17362,Philippines,2005,PHL,88015959.0,4.268641e+11,83.347,16.653,0.001,0.015,7.010,16.653,319.341,268.047,53.558,3628.220,-2.919,0.744,25.38,448.65,0.748,NaN,NaN,0.000000,0.000000,66.694,Low
22460,Vietnam,2005,VNM,81088309.0,3.089887e+11,87.842,12.158,0.000,0.000,12.115,12.158,369.791,322.285,44.605,4560.353,-6.920,8.722,24.09,462.56,1.197,NaN,NaN,0.000000,0.000000,75.684,Low
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11968,Malaysia,2024,MYS,35557670.0,NaN,91.718,8.282,0.624,0.000,6.094,8.282,1381.980,1262.247,113.976,38865.867,7.110,4.122,119.81,604.43,NaN,2.032771,-0.179576,284.315545,-6.261945,83.436,Medium
17381,Philippines,2024,PHL,115843663.0,NaN,87.182,12.818,1.408,0.482,4.130,12.818,652.563,572.515,84.178,5633.132,6.765,7.219,77.07,612.35,NaN,-0.804829,0.119433,-23.028884,4.601245,74.364,Low
18762,Singapore,2024,SGP,5832387.0,NaN,99.455,0.545,0.295,0.000,0.000,0.545,1053.921,1047.929,5.745,180701.453,19.774,6.255,29.73,498.74,NaN,10.547667,-0.052258,98.181818,-0.270745,98.910,High
20549,Thailand,2024,THA,71668010.0,NaN,92.597,7.403,0.935,0.582,1.104,7.403,1420.819,1312.641,104.940,19825.014,-0.549,2.597,110.68,554.73,NaN,-2.694532,0.221881,251.185958,-5.409022,85.194,Medium


In [28]:
year_bin = [2004, 2010, 2015, 2020, 2024] 
year_period = ['Period 1', 'Period 2', 'Period 3', 'Period 4']

sea_six_df['period'] = pd.cut(sea_six_df['year'], bins=year_bin, labels=year_period)
sea_six_df

,country,year,iso_code,population,gdp,fossil_share_energy,renewables_share_energy,solar_share_energy,wind_share_energy,hydro_share_energy,low_carbon_share_energy,primary_energy_consumption,fossil_fuel_consumption,renewables_consumption,energy_per_capita,renewables_cons_change_pct,fossil_cons_change_pct,greenhouse_gas_emissions,carbon_intensity_elec,energy_per_gdp,yoy_rse,yoy_fse,cum_rse,cum_fse,energy_gap,category,period
9739,Indonesia,2005,IDN,230871648.0,1.474875e+12,96.506,3.494,0.000,0.000,2.034,3.494,1416.892,1367.980,49.523,6137.142,6.250,2.824,86.45,650.54,0.961,NaN,NaN,0.000000,0.000000,93.012,Low,Period 1
11949,Malaysia,2005,MYS,25836075.0,3.997810e+11,97.845,2.155,0.000,0.000,1.933,2.155,836.425,817.333,17.999,32374.299,7.320,8.252,64.43,652.46,2.092,NaN,NaN,0.000000,0.000000,95.690,Medium,Period 1
18743,Singapore,2005,SGP,4268363.0,1.906915e+11,99.725,0.275,0.000,0.000,0.000,0.275,542.302,540.778,1.493,127051.617,-0.280,5.622,20.85,545.53,2.844,NaN,NaN,0.000000,0.000000,99.450,High,Period 1
17362,Philippines,2005,PHL,88015959.0,4.268641e+11,83.347,16.653,0.001,0.015,7.010,16.653,319.341,268.047,53.558,3628.220,-2.919,0.744,25.38,448.65,0.748,NaN,NaN,0.000000,0.000000,66.694,Low,Period 1
22460,Vietnam,2005,VNM,81088309.0,3.089887e+11,87.842,12.158,0.000,0.000,12.115,12.158,369.791,322.285,44.605,4560.353,-6.920,8.722,24.09,462.56,1.197,NaN,NaN,0.000000,0.000000,75.684,Low,Period 1
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
11968,Malaysia,2024,MYS,35557670.0,NaN,91.718,8.282,0.624,0.000,6.094,8.282,1381.980,1262.247,113.976,38865.867,7.110,4.122,119.81,604.43,NaN,2.032771,-0.179576,284.315545,-6.261945,83.436,Medium,Period 4
17381,Philippines,2024,PHL,115843663.0,NaN,87.182,12.818,1.408,0.482,4.130,12.818,652.563,572.515,84.178,5633.132,6.765,7.219,77.07,612.35,NaN,-0.804829,0.119433,-23.028884,4.601245,74.364,Low,Period 4
18762,Singapore,2024,SGP,5832387.0,NaN,99.455,0.545,0.295,0.000,0.000,0.545,1053.921,1047.929,5.745,180701.453,19.774,6.255,29.73,498.74,NaN,10.547667,-0.052258,98.181818,-0.270745,98.910,High,Period 4
20549,Thailand,2024,THA,71668010.0,NaN,92.597,7.403,0.935,0.582,1.104,7.403,1420.819,1312.641,104.940,19825.014,-0.549,2.597,110.68,554.73,NaN,-2.694532,0.221881,251.185958,-5.409022,85.194,Medium,Period 4


# CURRENT ANALYZATION

Based on the 2024 data, **Singapore** is undergoing the fastest renewable energy transition, recording a Year-over-Year (YoY) increase in renewable energy share of +10.55 percentage points this 2024 from -5.37% last 2023. This substantial rise indicates a rapid acceleration in the integration of renewable sources into its national energy mix within a single year.

In contrast, **Indonesia** leads in reducing fossil fuel dependence, posting the largest YoY decline in fossil energy share at −0.29 percentage points this 2024, from -0.54% last 2023, and +0.27% last 2006. Although the value is negative, it signifies the most aggressive reduction among the Southeast Asian countries in the dataset. This trend suggests that Indonesia is making the fastest progress in cleaning its energy grid relative to its regional peers.

Furthermore, the country that has transformed the most since 2005 is Malaysia, with a cumulative increase of 284% in their renewable energy share. This indicates a long-term structural shift in their national energy strategy, moving from a [Low/Medium] baseline to becoming a regional leader in adoption over the past two decades.

While the cumulative renewable share energy shows growth, cumumulative fossil share energy reveals the true retreat of traditional power. Vietnam is the leader in fossil fuel phase-out, having reduced their fossil share by a cumulative -10.67% since 2005. This demonstrates that their renewable growth is actively displacing carbon-intensive sources rather than just supplementing them.

Shockingly, the data for Philippines reveals a 'Carbon Regression' since 2005, characterized by a negative cumulative renewable share energy and a positive cumulative fossil share energy. This indicates that the country’s energy transition has not only stalled but reversed, with fossil fuels expanding their dominance in the energy mix. This likely stems from Economic Growth and Energy Security, positioning them as a primary outlier in the regional transition.

Moreover, in terms of their energy gap, where you will subtract the Fossil Share Energy to the Renewable Share energy, based on the calculated data, Vietnam is still the leader where it succesfully eradicated 19% of fossil fuel use in the country from 2005 up until 2024. Furthermore, Singapore on the otherhand remains the most fossil-dependent nation within the group. Its energy gap moved from 99.450% in 2005 to 98.910% in 2024, closing by only 0.54 percentage points. Its fossil fuel share barely budged, decreasing by just 0.27% over nearly two decades. 

Unlike its neighbors, the Philippines has seen its energy gap widen. In 2005, it had the smallest gap in the region at 66.694. By 2024, that gap grew to 74.364, an increase of 7.67 percentage points. This is because its fossil fuel share rose from 83.347% to 87.182% (+3.835%) while its renewable share actually dropped. With the provided data on the Philippines in relation to fossil and renewable share energy. This positions Philippines as the only country in the group moving backwards on energy transition, raising concerns about policy direction.


# PLOTLY VISUALIZATION

AAARGH

# 

In [29]:
color_map = {
    'Malaysia': '#F39C12',
    'Philippines': '#E74C3C',
    'Vietnam': '#27AE60',
    'Indonesia': '#3498DB',
    'Singapore': '#95A5A6',
    'Thailand': '#5DADE2'
}

fig = go.Figure()

countries = sea_six_df['country'].unique()

for country in countries:
    data = sea_six_df[sea_six_df['country'] == country].sort_values('year')
    fig.add_trace(go.Scatter(
        x=data['year'],
        y=data['renewables_share_energy'],
        name=country,
        mode='lines+markers',
        line=dict(color=color_map[country], width=3),
        marker=dict(size=8),
        visible=True
    ))

for country in countries:
    data = sea_six_df[sea_six_df['country'] == country].sort_values('year')
    fig.add_trace(go.Scatter(
        x=data['year'],
        y=data['fossil_share_energy'],
        name=country,
        mode='lines+markers',
        line=dict(color=color_map[country], width=3),
        marker=dict(size=8),
        visible=False
    ))

n = len(countries)

renewable_annotations = [
    dict(x=2006, y=19, text="Philippines Peak", showarrow=True, 
         arrowcolor="#E74C3C", ax=40, ay=-40, font=dict(color="#E74C3C", size=11)),
    dict(x=2018, y=6.5, text="Malaysia's Decline", showarrow=True, 
         arrowcolor="#F39C12", ax=40, ay=-30, font=dict(color="#F39C12", size=11)),
    dict(x=2015, y=28, text="Paris Agreement (2015)", showarrow=False, 
         font=dict(color="#2C3E50", size=10))
]

fossil_annotations = [
    dict(x=2006, y=81, text="Philippines Peak", showarrow=True, 
         arrowcolor="#E74C3C", ax=-40, ay=40, font=dict(color="#E74C3C", size=11)),
    dict(x=2018, y=94, text="Malaysia's Decline", showarrow=True, 
         arrowcolor="#F39C12", ax=40, ay=-30, font=dict(color="#F39C12", size=11)),
    dict(x=2015, y=75, text="Paris Agreement (2015)", showarrow=False, 
         font=dict(color="#2C3E50", size=10))
]

fig.update_layout(
    updatemenus=[dict(
        buttons=[
            dict(
                label="Renewable Energy Share",
                method="update",
                args=[
                    {"visible": [True]*n + [False]*n},
                    {
                        "yaxis.title": "Renewable Share (%)",
                        "title.text": "Southeast Asia Renewable Energy Transition (2005-2024)",
                        "yaxis.range": [0, 30],
                        "annotations": renewable_annotations
                    }
                ]
            ),
            dict(
                label="Fossil Fuels Share",
                method="update",
                args=[
                    {"visible": [False]*n + [True]*n},
                    {
                        "yaxis.title": "Fossil Fuel Share (%)",
                        "title.text": "Southeast Asia Fossil Fuel Dependence (2005-2024)",
                        "yaxis.range": [70, 105],
                        "annotations": fossil_annotations
                    }
                ]
            )
        ],
        x=0.12,
        y=1.15,
        bgcolor="white",
        bordercolor="#2C3E50",
        borderwidth=2
    )]
)

fig.update_layout(
    title="Southeast Asia Renewable Energy Transition (2005-2024)",
    title_x=0.5,
    title_font_size=20,
    
    plot_bgcolor='white',
    paper_bgcolor='#F8F9FA',
    
    xaxis=dict(
        title="Year", 
        showgrid=True, 
        gridcolor='#E8E8E8', 
        dtick=2,
        range=[2004.5, 2024.5]
    ),
    yaxis=dict(
        title="Renewable Share (%)", 
        showgrid=True, 
        gridcolor='#E8E8E8',
        range=[0, 30]
    ),
    
    legend=dict(
        title="Country", 
        x=1.02, 
        y=1, 
        bgcolor='white', 
        bordercolor='#2C3E50', 
        borderwidth=1
    ),
    
    height=600,
    width=1100,
    margin=dict(l=80, r=150, t=120, b=80),
    
    annotations=renewable_annotations,
    hovermode='closest'
)

fig.add_vline(x=2015, line_dash="dot", line_color="#7F8C8D", opacity=0.5)

fig.show()

In [30]:
df_2024 = sea_six_df[sea_six_df['year'] == 2024].copy()

df_2024 = df_2024.sort_values('cum_rse', ascending=False)

colors = ['#E74C3C' if x < 0 else '#27AE60' for x in df_2024['cum_rse']]

df_2005 = sea_six_df[sea_six_df['year'] == 2005][['country', 'renewables_share_energy']].rename(
    columns={'renewables_share_energy': 'baseline_2005'}
)
df_2024 = df_2024.merge(df_2005, on='country', how='left')

fig = go.Figure()

fig.add_trace(go.Bar(
    x=df_2024['country'],
    y=df_2024['cum_rse'],
    orientation='v',
    marker=dict(
        color=colors,
        line=dict(color='#2C3E50', width=1)
    ),
    text=[f"{val:+.1f}%" for val in df_2024['cum_rse']],
    textposition='outside',
    textfont=dict(size=12, color='#2C3E50'),
    hovertemplate="<b>%{x}</b><br>" +
                  "Cumulative Change: %{y:+.2f}%<br>" +
                  "2005 Baseline: %{customdata[0]:.2f}%<br>" +
                  "2024 Current: %{customdata[1]:.2f}%<br>" +
                  "<extra></extra>",
    customdata=df_2024[['baseline_2005', 'renewables_share_energy']].values
))

# Layout
fig.update_layout(
    title={
        'text': "Energy Transformation Champions: Cumulative Change Since 2005<br>" +
            "<sub>Percentage change in renewable energy share of total energy consumption</sub>",
        'x': 0.5,
        'xanchor': 'center',
        'font': {'size': 20, 'color': '#2C3E50', 'family': 'Arial Black'}
    },
    
    xaxis=dict(
        title="",
        title_font_size=14,
        showgrid=False
    ),
    
    yaxis=dict(
        title="Cumulative % Change in Renewable Energy Share",
        title_font_size=14,
        showgrid=True,
        gridcolor='#E8E8E8',
        zeroline=True,
        zerolinecolor='#2C3E50',
        zerolinewidth=2,
        range=[-50, None]
    ),
    
    plot_bgcolor='white',
    paper_bgcolor='#F8F9FA',
    
    height=600,
    width=1000,
    margin=dict(l=100, r=100, t=120, b=80),
    hovermode='closest'
)

# Add horizontal line at zero
fig.add_hline(y=0, line_color='#2C3E50', line_width=2)

fig.show()

In [31]:
import plotly.graph_objects as go
import pandas as pd

# Filter for 2024 data only
df_2024 = sea_six_df[sea_six_df['year'] == 2024].copy()

# Get 2005 baseline values for hover
df_2005 = sea_six_df[sea_six_df['year'] == 2005][['country', 'renewables_share_energy']].rename(
    columns={'renewables_share_energy': 'baseline_2005'}
)
df_2024 = df_2024.merge(df_2005, on='country', how='left')

# Sort by cumulative change (DESCENDING for highest at top)
df_sorted_by_value = df_2024.sort_values('cum_rse', ascending=False)
df_sorted_alphabetically = df_2024.sort_values('country', ascending=True)

# Color scheme
def get_colors(df):
    colors = []
    for idx, row in df.iterrows():
        if row['cum_rse'] < 0:
            colors.append('#E74C3C')  # Red for negative
        elif row['cum_rse'] > 200:
            colors.append('#27AE60')  # Dark green for >200%
        elif row['cum_rse'] > 150:
            colors.append('#2ECC71')  # Medium green for 150-200%
        else:
            colors.append('#58D68D')  # Light green for <150%
    return colors

# Create figure
fig = go.Figure()

# Trace 1: Sorted by value (visible by default)
fig.add_trace(go.Bar(
    y=df_sorted_by_value['country'],
    x=df_sorted_by_value['cum_rse'],
    orientation='h',
    marker=dict(
        color=get_colors(df_sorted_by_value),
        line=dict(color='#2C3E50', width=1.5)
    ),
    # FIXED: Round to 2 decimals, better formatting
    text=[f"+{val:.2f}%" if val > 0 else f"{val:.2f}%" for val in df_sorted_by_value['cum_rse']],
    textposition='outside',
    textfont=dict(size=14, color='#2C3E50', family='Arial', weight='bold'),
    hovertemplate="<b>%{y}</b><br>" +
                  "<b>Cumulative Change:</b> %{x:.2f}%<br>" +
                  "2005 Baseline: %{customdata[0]:.2f}%<br>" +
                  "2024 Current: %{customdata[1]:.2f}%<br>" +
                  "<extra></extra>",
    customdata=df_sorted_by_value[['baseline_2005', 'renewables_share_energy']].values,
    visible=True
))

# Trace 2: Sorted alphabetically (hidden by default)
fig.add_trace(go.Bar(
    y=df_sorted_alphabetically['country'],
    x=df_sorted_alphabetically['cum_rse'],
    orientation='h',
    marker=dict(
        color=get_colors(df_sorted_alphabetically),
        line=dict(color='#2C3E50', width=1.5)
    ),
    text=[f"+{val:.2f}%" if val > 0 else f"{val:.2f}%" for val in df_sorted_alphabetically['cum_rse']],
    textposition='outside',
    textfont=dict(size=14, color='#2C3E50', family='Arial', weight='bold'),
    hovertemplate="<b>%{y}</b><br>" +
                  "<b>Cumulative Change:</b> %{x:.2f}%<br>" +
                  "2005 Baseline: %{customdata[0]:.2f}%<br>" +
                  "2024 Current: %{customdata[1]:.2f}%<br>" +
                  "<extra></extra>",
    customdata=df_sorted_alphabetically[['baseline_2005', 'renewables_share_energy']].values,
    visible=False
))

# Improved dropdown - smaller, cleaner
fig.update_layout(
    updatemenus=[dict(
        buttons=[
            dict(
                label="Highest First",
                method="update",
                args=[{"visible": [True, False]}]
            ),
            dict(
                label="Alphabetical",
                method="update",
                args=[{"visible": [False, True]}]
            )
        ],
        x=0.01,
        y=1.05,
        xanchor='left',
        yanchor='top',
        bgcolor="#FFFFFF",
        bordercolor="#95A5A6",
        borderwidth=1,
        font=dict(size=11, color='#2C3E50')
    )]
)

# Layout - NO SUBTITLE, cleaner spacing
fig.update_layout(
    title={
        'text': "Energy Transformation Champions: Cumulative Change Since 2005",
        'x': 0.5,
        'xanchor': 'center',
        'font': {'size': 22, 'color': '#2C3E50', 'family': 'Arial Black'}
    },
    
    xaxis=dict(
        title="Cumulative % Change",
        title_font_size=14,
        title_font_color='#2C3E50',
        showgrid=True,
        gridcolor='#E8E8E8',
        gridwidth=1,
        zeroline=True,
        zerolinecolor='#34495E',
        zerolinewidth=2.5,
        showline=True,
        linewidth=2,
        linecolor='#2C3E50',
        tickfont=dict(size=12),
        dtick=50
    ),
    
    yaxis=dict(
        title="",
        showgrid=False,
        showline=False,
        tickfont=dict(size=14, color='#2C3E50', family='Arial')
    ),
    
    plot_bgcolor='white',
    paper_bgcolor='#F8F9FA',
    
    # INCREASED height for better spacing
    height=550,
    width=1100,
    # INCREASED margins - especially right for labels
    margin=dict(l=120, r=200, t=100, b=80),
    
    hovermode='closest',
    showlegend=False
)

# Thicker zero line
fig.add_vline(x=0, line_color='#34495E', line_width=2.5, opacity=0.8)

fig.show()

In [32]:
# yoy_rse	yoy_fse	cum_rse	cum_fse
random = sea_six_df.pivot_table(index='period', columns='country', values='yoy_rse').mean()
random

/tmp/ipykernel_17/2185168811.py:2: FutureWarning:

The default value of observed=False is deprecated and will change to observed=True in a future version of pandas. Specify observed=False to silence this warning and retain the current behavior



country
Indonesia      7.154037
Malaysia       7.672371
Philippines   -1.087120
Singapore      4.958673
Thailand       6.944529
Vietnam        4.672842
dtype: float64

In [33]:
avg_yoy_table = sea_six_df.pivot_table(
    index='country',
    columns='period',
    values='yoy_rse',
    aggfunc='mean'
)

avg_yoy_table

/tmp/ipykernel_17/1828639748.py:1: FutureWarning:

The default value of observed=False is deprecated and will change to observed=True in a future version of pandas. Specify observed=False to silence this warning and retain the current behavior



period,Period 1,Period 2,Period 3,Period 4
country,,,,
Indonesia,5.122239,-0.654713,23.295042,0.853580
Malaysia,1.101238,12.006617,13.914468,3.667160
Philippines,-0.665792,-1.529818,-2.138660,-0.014208
Singapore,-3.138010,4.670226,1.562799,16.739678
Thailand,6.130192,9.239559,10.850129,1.558235
Vietnam,4.050737,6.625644,1.096399,6.918589
